packages/apps/reputation-oracle/server/src/common/guards/signature.auth.spec.ts

```diff
@@ -79,18 +78,34 @@ describe('SignatureAuthGuard', () => {
     });
 
     it('should throw unauthorized exception if signature is not verified', async () => {
-      (verifySignature as jest.Mock).mockReturnValue(false);
+      (verifySignature as jest.Mock).mockReturnValueOnce(false);
 
-      await expect(guard.canActivate(context as any)).rejects.toThrow(
-        new ControlledError('Unauthorized', HttpStatus.UNAUTHORIZED),
-      );
+      try {
+        await guard.canActivate(context as any);
+      } catch (error) {
+        expect(error).toBeInstanceOf(HttpException);
+        expect(error.response).toHaveProperty(
+          'message',
+          'Invalid web3 signature',
+        );
+        expect(error.response).toHaveProperty('timestamp');
+        expect(error).toHaveProperty('status', HttpStatus.UNAUTHORIZED);
+      }
     });
 
     it('should throw unauthorized exception for unrecognized oracle type', async () => {
       mockRequest.originalUrl = '/some/random/path';
-      await expect(guard.canActivate(context as any)).rejects.toThrow(
-        new ControlledError('Unauthorized', HttpStatus.UNAUTHORIZED),
-      );
+      try {
+        await guard.canActivate(context as any);
+      } catch (error) {
+        expect(error).toBeInstanceOf(HttpException);
```

Comment:
If you do testing using `try..catch` block, you should do something like:
```
let thrownError: any;
try {
  await someService.logicToTest();
} catch (error) {
  thrownError = error;
}

expect(thrownError).toBeDefined();
expect(thrownError.message).toBe('expected message');
```

Otherwise there is a risk that the method you test executes successfully and test case silently passes